Add detailed documentation for standard Grails application profiles

jamesfredley opened the pull request:
Introduces new documentation sections and files for each standard Grails application profile, including web, REST API, plugin, web-plugin, rest-api-plugin, and profile profiles.

Resolves:  https://github.com/apache/grails-core/issues/14695

jamesfredley committed:
Add detailed documentation for default Grails application profiles

Introduces new documentation sections and files for each default Grails application profile, including web, REST API, plugin, web-plugin, rest-api-plugin, and profile profiles. Updates the guide index and table of contents to reference these new sections, providing users with comprehensive information on available profiles, their commands, and features.

jamesfredley committed:
Rename 'Default Profiles' to 'Standard Profiles' in docs

Updated documentation to refer to 'Standard Profiles' instead of 'Default Profiles'. Renamed the relevant AsciiDoc file, updated references in the index and table of contents, and adjusted terminology for consistency.

sbglasius reviewed (approved):
As a developer perhaps it would be nice to elaborate over the `profile` profile. I for one don't know  how to use it 🙂

matrei commented on grails-doc/src/en/guide/profiles/standardProfiles.adoc:
```diff
@@ -0,0 +1,58 @@
+
+[source,bash]
+----
+$ grails list-plugins
```
`list-profiles`?

jamesfredley replied:
Fixed

dauer commented:
@jamesfredley the grails package-plugin is mentioned in the documentation so I guess this https://github.com/apache/grails-core/issues/11613 is no longer relevant.
I tested creating a Grails 4.0.4 plugin and I have the package-plugin command, I am not sure it is a Grails issue, probably as missing plugin or dependency in his project....

This is probaly not relevant any more
https://github.com/apache/grails-core/pull/14091

matrei reviewed (commented):
The section `6.4 Publishing Profiles` (which is outside this PR) has this snippet:
```groovy
grailsPublish {
        githubSlug = 'foo/bar'
        license {
            name = 'Apache-2.0'
        }
        title = 'My plugin title'
        desc = 'My plugin description'
        developers = [johndoe: 'John Doe']
    }
```
which should probably be:
```groovy
grailsPublish {
    githubSlug = 'foo/bar'
    license {
        name = 'Apache-2.0'
    }
    title = 'My profile title'
    desc = 'My profile description'
    developers = [johndoe: 'John Doe']
}
```

jamesfredley committed:
Correct command from 'list-plugins' to 'list-profiles'

jamesfredley commented:
https://github.com/apache/grails-core/issues/11613 and https://github.com/apache/grails-core/pull/14091 have been updated and closed given package-plugin is available for projects with grails profiles plugin, web-plugin and rest-api-plugin.

jamesfredley commented:
@sbglasius Do you think 6.1 - 6.6 on https://docs.grails.org/snapshot/guide/profiles.html#creatingProfiles cover the `profile` profile well or should we create an issue to add further improvements?

jamesfredley committed:
Update example title and desc in publishingProfiles guide

Changed example values for 'title' and 'desc' from 'My plugin title/description' to 'My profile title/description' to better reflect the context of profile publishing.

jamesfredley commented:
section 6.4 Publishing Profiles has been updated
